src/transaction/log_page_buffer.c

```diff
@@ -7303,6 +7304,97 @@ logpb_checkpoint (THREAD_ENTRY * thread_p)
 #undef detailed_er_log
 }
 
+/*
+ * log_checkpoint_trantable - execute a transaction table checkpoint;
+ *          wait for log pages to be persisted up to the lsa where where the transaction table
+ *          snapshot was taken; delete previous transaction table checkpoints
+ *
+ * return: error status
+ *
+ */
+int
+logpb_checkpoint_trantable (THREAD_ENTRY * const thread_p)
+{
+  const bool detailed_logging = prm_get_bool_value (PRM_ID_LOG_CHKPT_DETAILED);
+
+  if (!is_tran_server_with_remote_storage ())
+    {
+      er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: only allowed on transaction server with remote storage\n");
+      return ER_FAILED;
+    }
+
+  {
+    LOG_CS_ENTER (thread_p);
+    // *INDENT-OFF*
+    scope_exit <std::function<void (void)>> unlock_log_cs_on_exit ([thread_p] ()
+    {
+      LOG_CS_EXIT(thread_p);
+    });
+    // *INDENT-ON*
+
+    cublog::checkpoint_info trantable_checkpoint_info;
+
+    if (detailed_logging)
+      {
+	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: started, loading trantable\n");
+      }
+    LOG_LSA dummy_smallest_tran_lsa = NULL_LSA;
+    trantable_checkpoint_info.load_trantable_snapshot (thread_p, dummy_smallest_tran_lsa);
+
+    // loading the transaction table snapshot ensures also that a snapshot lsa has been set
+    const log_lsa trantable_checkpoint_lsa = trantable_checkpoint_info.get_snapshot_lsa ();
+
+    if (trantable_checkpoint_lsa.is_null ())
+      {
+	// no transaction table checkpoint can be recorded
+	er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: failed; null checkpoint lsa\n");
+	return ER_FAILED;
+      }
+
+    if (detailed_logging)
+      {
+	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: adding with lsa=%lld|%d\n",
+		       LSA_AS_ARGS (&trantable_checkpoint_lsa));
+      }
+    log_Gl.m_metainfo.add_checkpoint_info (trantable_checkpoint_lsa, std::move (trantable_checkpoint_info));
+
+    // make sure new checkpoint is persisted to disk
+    const int res_metalog_to_file = log_write_metalog_to_file ();
+    if (res_metalog_to_file != NO_ERROR)
+      {
+	ASSERT_ERROR ();
+	er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: failed; writing metalog to file\n");
```

Comment:
Always log debilitating errors
```suggestion
	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: failed; writing metalog to file\n");
```